src/nickel.rs

```diff
@@ -190,6 +190,58 @@ impl<D: Sync + Send + 'static> Nickel<D> {
     }
 }
 
+#[cfg(feature = "ssl")]
+mod ssl {
+    use std::net::ToSocketAddrs;
+    use std::env;
+    use server::Server;
+    use hyper::status::StatusCode;
+    use hyper::net::Ssl;
+
+    use super::Nickel;
+
+    /// Bind and listen for connections on the given host and port.
+    /// Only accepts SSL connections
+    ///
+    /// # Panics
+    ///
+    /// Panics if `addr` is an invalid `SocketAddr`.
+    ///
+    /// # Examples
+    /// ```{rust,no_run}
+    /// use nickel::Nickel;
+    /// use hyper::net::Openssl;
+    ///
+    /// let server = Nickel::new();
+    /// let ssl = Openssl::with_cert_and_key("foo.crt", "key.pem").unwrap();
+    /// server.listen_https("127.0.0.1:6767", ssl);
+    /// ```
+    impl <D: Sync + Send + 'static> Nickel<D> {
+        pub fn listen_https<T,S>(mut self, addr: T, ssl: S)
+        where T: ToSocketAddrs,
+              S: Ssl + Sync + Send + Clone + 'static {
```

Comment:
Is `Sync` required here? (corresponding api in [hyper](https://github.com/hyperium/hyper/blob/master/src/server/mod.rs#L217) requires only `Ssl + Clone + Send`)